external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -140,84 +126,706 @@ struct PackageWithResolverTOML<'a>(&'a Package);
+        for pkg_name in all_packges {
+            let mut existing_pkg_info: Option<(Symbol, &DependencyGraph, &Package)> = None;
+            for (dep_name, (g, _)) in &dep_graphs {
+                if let Some(pkg) = g.package_table.get(&pkg_name) {
+                    // graph g has a package with name pkg_name
+                    if let Some((existing_dep_name, existing_graph, existing_pkg)) =
+                        existing_pkg_info
+                    {
+                        // it's the subsequent time package with pkg_name has been encountered
+                        if pkg == existing_pkg {
+                            // both packages are the same but we need to check if their dependencies
+                            // are the same as well
+                            let (exisiting_pkg_deps, pkg_deps) =
+                                deps_equal(pkg_name, &existing_graph, &g);
+                            if exisiting_pkg_deps != pkg_deps {
+                                bail!(
+                                    "When resolving dependencies for package {}, \
+                                     conflicting dependencies found for '{}':\n{}{}",
+                                    root_package,
+                                    pkg_name,
+                                    format_deps(existing_dep_name, exisiting_pkg_deps),
+                                    format_deps(*dep_name, pkg_deps),
+                                )
+                            }
+
+                            continue;
+                        } else {
+                            bail!(
+                                "When resolving dependencies for package {0}, \
+                                 conflicting dependencies found:\nin {4}: {1} = {2}\nin {5}: {1} = {3}",
+                                root_package,
+                                pkg_name,
+                                PackageWithResolverTOML(existing_pkg),
+                                PackageWithResolverTOML(&pkg),
+                                existing_dep_name,
+                                dep_name,
+                            );
+                        }
+                    } else {
+                        // first time this package was encountered
+                        existing_pkg_info = Some((*dep_name, g, pkg));
+                    }
+                }
+            }
+            if let Some((_, g, existing_pkg)) = existing_pkg_info {
+                // update combined graph with the new package and its dependencies
+                self.package_table.insert(pkg_name, existing_pkg.clone());
+                for to_pkg_name in g
+                    .package_graph
+                    .neighbors_directed(pkg_name, Direction::Outgoing)
+                {
+                    // unwrap is safe as all edges have a Dependency weight
+                    let sub_dep = g.package_graph.edge_weight(pkg_name, to_pkg_name).unwrap();
```

Comment:
nit: uses `edges_directed` to avoid having to `unwrap` in the first place.

Reply:
I am not sure if you can call `edges_directed` on a graphmap but `edges` works here just as well. Thanks!